### DISCUSSION TITLE
Walkthrough: Stream status

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#543] [Show and tell]
I made this diagram that highlights the allowed transitions between stream statuses. This diagram is also available in our [docs](https://docs.sablier.com/concepts/lockup/statuses).

```mermaid
---
title: Stream Status
---
stateDiagram-v2
  direction LR
  state Warm {
    Pending
    Streaming
  }
  state Cold {
    Settled
    Canceled
    Depleted
  }
  Null --> Pending
  Null --> Settled
  Pending --> Streaming
  Pending --> Settled
  Pending --> Canceled
  Streaming --> Settled
  Streaming --> Canceled
  Streaming --> Depleted
  Settled --> Depleted
  Canceled --> Depleted
```

**Comment by PaulRBerg** [comment]
Note: this diagram is now part of the docs website:

https://docs.sablier.com/concepts/protocol/statuses

**Comment by smol-ninja** [comment]
Thanks for the making this diagram. It is really cool. I am closing this discussion now as it has already been shipped with the docs.

**Reply by PaulRBerg** [reply]
Thanks! But why close it? I think that "Show and Tell"-like discussions are helpful for external readers of this repository. They might not know that the status is documented on our docs site.
